portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -76,97 +70,88 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       blogMap.putInMap(messages);
 
-      // Load messages from BlogHashMap and respond with gson.
-      List<String> tagsToSearch = new ArrayList<String>();
-      tagsToSearch.add(""); // we'll get inputs later.
-
+       // Load messages from BlogHashMap and respond with gson.
       LinkedList<BlogMessage> loadedBlogMessages = blogMap.getMessages(tagsToSearch, numberOfCommentsToDisplay);
     
       Gson gson = new Gson();
       response.setContentType("application/json;");
       
       response.getWriter().println(gson.toJson(loadedBlogMessages));
       return;
-
     }
     
-
-    /**
-   * Converts a ServerStats instance into a JSON string using the Gson library
-   */
-    @Override
+   @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      // Get Post parameters.
       String message = request.getParameter("text-input");
-
-      String sender = getParameter(request, "sender", "Steven");
+      String sender = request.getParameter("sender");
+      String loadFactor = request.getParameter("load_factor");
+      String postTag = getTagParameter(request, "tags", InternalTags.defaultTag());
       
-      // TODO: 
-      //      Get default tag from the InternalTags class and use that below.
+      //TODO: Handle image file sent with FormData.
+      //String imageUrl = getUploadedFileUrl(request, imagePath);
       
-      // Get type of comment.
-      String commentType = getParameter(request, "tags", "Default");
-
-      //String imageUrl = getUploadedFileUrl(request, "image");
+      // Only put BlogMessages with a message and tag in datastore.
+      if (message.equals("") || message.equals(null) || postTag == null) {
+        return;
+      }
 
-      String messageRepliesString = getParameter(request, "replies", "");
-	    String messageRepliesArray[] = messageRepliesString.split(",");
-	    List<String> messageReplies = new ArrayList<String>();
-	    messageReplies = Arrays.asList(messageRepliesArray);
-      
       long timestamp = System.currentTimeMillis();
-
       Entity blogMessageEntity = new Entity("blogMessage");
       blogMessageEntity.setProperty("sender", sender);
       blogMessageEntity.setProperty("text", message);
       //blogMessageEntity.setProperty("imgUrl", imageUrl);
       blogMessageEntity.setProperty("time", timestamp);
-      blogMessageEntity.setProperty("tag", commentType);
+      blogMessageEntity.setProperty("tag", postTag);
       blogMessageEntity.setProperty("replies", messageReplies);
+    
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       datastore.put(blogMessageEntity);
 
-      response.sendRedirect("/index.html");
+      // To get the recently posted message, add its tag to the |tagsToSearch| list.
+      // If the list has tags already, clear them before adding the tag.
+      // Our goal is to get the recently posted message.
+      if (!tagsToSearch.isEmpty()) {
```

Comment:
For lines 126-128, consider just call tagsToSearch.clear() without the conditional check